File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/layout/LayoutViewComponent.java
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.layout;

import de.tschuehly.spring.viewcomponent.core.IViewContext;
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;

@ViewComponent
public
class LayoutViewComponent {

    private record LayoutView(IViewContext nestedViewComponent) implements IViewContext{}

    public IViewContext render(IViewContext nestedViewComponent) {
        return new LayoutView(nestedViewComponent);
    }
}

File: examples/thymeleaf-kotlin-example/src/main/kotlin/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/layout/LayoutViewComponent.kt
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.layout

import de.tschuehly.spring.viewcomponent.core.IViewContext
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent

@ViewComponent
class LayoutViewComponent {
    data class LayoutView(val nestedViewComponent: IViewContext) : IViewContext

    fun render(nestedViewComponent: IViewContext) = LayoutView(nestedViewComponent)

}